src/Neo/SmartContract/Native/CryptoLib.cs

```diff
@@ -83,5 +89,82 @@ public static bool VerifyWithECDsa(byte[] message, byte[] pubkey, byte[] signatu
+        /// <param name="g2">Gt1 point as byteArray</param>
+        /// <returns></returns>
+        [ContractMethod(CpuFee = 1 << 19)]
+        [RequiresPreviewFeaturesAttribute]
+        public static byte[] Bls12381Add(byte[] g1, byte[] g2)
+        {
+            if (g1.Length != g2.Length)
+                throw new Exception($"Bls12381 operation fault, type:format, error:type mismatch");
+            byte[] result;
+            switch (g1.Length)
+            {
+                case G1:
+                    result = new G1Affine(new G1Projective(G1Affine.FromCompressed(g1)) + new G1Projective(G1Affine.FromCompressed(g2))).ToCompressed();
+                    break;
+                case G2:
+                    result = new G2Affine(new G2Projective(G2Affine.FromCompressed(g1)) + new G2Projective(G2Affine.FromCompressed(g2))).ToCompressed();
+                    break;
+                case Gt:
+                    result = (Cryptography.BLS12_381.Gt.FromBytes(g1) + Cryptography.BLS12_381.Gt.FromBytes(g2)).ToArray();
+                    break;
+                default:
+                    throw new Exception($"Bls12381 operation fault, type:format, error:valid point length");
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// The implementation of System.Crypto.PointMul.
+        /// Mul operation of gt point and mulitiplier
+        /// </summary>
+        /// <param name="g">Gt point as byteArray</param>
+        /// <param name="mul">Mulitiplier</param>
+        /// <returns></returns>
+        [ContractMethod(CpuFee = 1 << 21)]
+        [RequiresPreviewFeaturesAttribute]
+        public static byte[] Bls12381Mul(byte[] g, long mul)
+        {
+            Scalar X = mul < 0 ? -new Scalar(Convert.ToUInt64(Math.Abs(mul))) : new Scalar(Convert.ToUInt64(Math.Abs(mul)));
+            byte[] result;
+            switch (g.Length)
+            {
+                case G1:
+                    result = new G1Affine(G1Affine.FromCompressed(g) * X).ToCompressed();
+                    break;
+                case G2:
+                    result = new G2Affine(G2Affine.FromCompressed(g) * X).ToCompressed();
+                    break;
+                case Gt:
+                    result = (Cryptography.BLS12_381.Gt.FromBytes(g) * X).ToArray();
+                    break;
+                default:
+                    throw new Exception($"Bls12381 operation fault, type:format, error:valid point length");
+            }
+            return result;
+        }
+
+        /// <summary>
+        /// The implementation of System.Crypto.PointPairing.
+        /// Pairing operation of g1 and g2
+        /// </summary>
+        /// <param name="g1">Gt point1 as byteArray</param>
+        /// <param name="g2">Gt point2 as byteArray</param>
+        /// <returns></returns>
+        [ContractMethod(CpuFee = 1 << 23)]
+        [RequiresPreviewFeaturesAttribute]
```

Comment:
It's required?

Reply:
I tried to remove it, but I get an error